src/main/scala/wdl4s/MemorySize.scala

```diff
@@ -0,0 +1,39 @@
+package wdl4s
+
+import wdl4s.parser.MemoryUnit
+
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+
+object MemorySize {
+  val memoryPattern = """(\d+(?:\.\d+)?)\s*(\w+)""".r
+  def parse(unparsed: String): Try[MemorySize] = {
```

Comment:
Weren't you were one of the people early on who was pushing really hard for `Try`s *instead* of throwing exceptions?  Don't you remember those arguments we had way back when where you argued that it's better to encapsulate the failure in the return type?  I'm sure @mcovarr remembers that too!